### PR TITLE
HPCC-27101 Roxie Error reading query from socket Maximum block size

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Smoke test and manual Roxie testing

### CODE REVIEW

**Comment by mckellyln** [comment]
Or should we save each protocolPlugin and free them at Roxie shutdown ?

**Comment by mckellyln** [comment]
@richardkchapman would like to get this fix into next 8.6.0 rc

**Review by ghalliday — APPROVED** [review]
Ugly, but probably the safest way to fix the regression.  Is there another jira to investigate and fix it properly?

File: roxie/ccd/ccdmain.cpp [hunk 1]
```diff
@@ -1397,7 +1397,8 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
                         }
                         const char *soname =  roxieFarm.queryProp("@so");
                         const char *config  = roxieFarm.queryProp("@config");
-                        Owned<IHpccProtocolPlugin> protocolPlugin = ensureProtocolPlugin(*protocolCtx, soname);
+                        // NB: leaks, but cannot be Owned<> until we solve scope properly
```
Comment (richardkchapman) [inline roxie/ccd/ccdmain.cpp:1400]: It's really "until we fix the bug in ensureProtocolPlugin whereby some paths return a linked object and others do not."

Comment (afishbeck) [reply roxie/ccd/ccdmain.cpp]: There is a JIRA for the proper fix:  https://track.hpccsystems.com/browse/HPCC-25664

I can put together a fix, unless @mckellyln wants to look at it.

**Review by afishbeck — APPROVED** [review]

**Pull request merged**